src/App.js

```diff
@@ -1,79 +1,127 @@
 import React from 'react';
 
-import { getAll, getById } from './api/phone'
+import {getAll, getById} from './api/phone'
 import Basket from './Basket'
 import Filter from './Filter'
 import Catalog from './Catalog'
-
+import Viewer from './Viewer'
 import './App.css';
 
 
 class App extends React.Component {
-  constructor(props) {
-    super(props);
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            phones: getAll(),
+            selectedPhone: null,
+            basketItems: [],
+        };
+    }
+
+    addToBasket = (phoneId) => {
+        const items = this.state.basketItems;
```

Comment:
Can't understand why do you need to reassign link to `basketItems` to another variable.
```suggestion
        const {basketItems} = this.state;
```
This is similar to what you have done.

Reply:
I'd recommend having a copy of items instead.
```javascript
const items = [...this.state.basketItems];
```
It will help you to avoid state mutation